### ISSUE TITLE
Fix S2589 FP: Constants not correctly evaluated for S2589

### ISSUE THREAD

**Issue by nrankin18** [SonarSource/sonar-dotnet#8512]
### Description

S2589 does not correctly evaluate constant and considers code to be unreachable.

### Repro steps

```cs
private const int SOME_CONST = 3;

private void Test(List<string> elements)
{
	int i = 0;
	foreach (string s in elements) {
		i++;
		if (i >= SOME_CONST) { // S2589 is thrown flagging the conditional as always false
			Console.WriteLine("Here");
		}
	}
}
```

### Expected behavior

S2589 should not be raised because the conditional is passable when `elements.Count >= SOME_CONST`

### Actual behavior

S2589 is thrown.

### Known workarounds

None.

### Related information

* Related information
* C# version: 10.0
* Visual Studio version: 17.8.2
* dotnet version : net6.0
* Operating System: Windows 11 Home 10.0.22000 Build 22000

**Comment by Tim-Pohlmann** [comment]
Hello @nrankin18, thank you for the report. The issue is related to loop handling. It will be fixed by #8474.
